pkg/controller/stackconfigpolicy/controller.go

```diff
@@ -314,23 +332,41 @@ func (r *ReconcileStackConfigPolicy) reconcileElasticsearchResources(ctx context
 			return results.WithError(err), status
 		}
 
-		// check that there is no other policy that already owns the Settings Secret
-		currentOwner, ok := filesettings.CanBeOwnedBy(actualSettingsSecret, policy)
-		if !ok {
-			err = fmt.Errorf("conflict: resource Elasticsearch %s/%s already configured by StackConfigpolicy %s/%s", es.Namespace, es.Name, currentOwner.Namespace, currentOwner.Name)
-			r.recorder.Eventf(&policy, corev1.EventTypeWarning, events.EventReasonUnexpected, err.Error())
-			results.WithError(err)
-			err = status.AddPolicyErrorFor(esNsn, policyv1alpha1.ConflictPhase, err.Error(), policyv1alpha1.ElasticsearchResourceType)
+		// build the final config by merging all policies that target the given Elasticsearch cluster
+		esPolicyConfigFinal, err := getPolicyConfigForElasticsearch(&es, allPolicies, r.params)
+		switch {
+		case errors.Is(err, errMergeConflict):
+			log.V(1).Info("StackConfigPolicy merge conflict for Elasticsearch", "es_namespace", es.Namespace, "es_name", es.Name, "error", err)
+			results.WithRequeue(defaultRequeue)
+			if esPolicyConfigFinal == nil {
+				continue
+			}
+			conflictErr, exists := esPolicyConfigFinal.PoliciesWithConflictErrors[reconcilingPolicyNsn]
```

Comment:
Let's maybe add a comment to explain why using only `err` is not enough,  and why we need to get `conflictErr` from `esPolicyConfigFinal.PoliciesWithConflictErrors`